core/parsigdb/memory.go

```diff
@@ -87,6 +87,7 @@ func (db *MemDB) StoreInternal(ctx context.Context, duty core.Duty, signedSet co
 
 // StoreExternal stores an externally received partially signed duty data set.
 func (db *MemDB) StoreExternal(ctx context.Context, duty core.Duty, signedSet core.ParSignedDataSet) error {
+	thresholdReached := make(map[core.PubKey]bool)
```

Comment:
this doesn't solve the problem you have in the logs, please write tests to verify your logic is correct. Note that you have two peers submitting/storing psigs sequentially.

Reply:
that is the problem, first is reaching the threshold which then proceeds with signature aggregation and then othen peer after that exceeds the threshold which also proceeds with signature aggregation and then bcast.

i have reverted my change and added a test to demonstrate the problem.

Reply:
Added test and corrected implementation